inspirehep/modules/workflows/tasks/upload.py

```diff
@@ -43,34 +45,106 @@ def store_record(obj, eng):
     """Insert or replace a record."""
     is_update = obj.extra_data.get('is-update')
     is_authors = eng.workflow_definition.data_type == 'authors'
+    if not current_app.config.get("ENABLE_INSPIREHEP_REMOTE_RECORD_MANAGEMENT"):
+        with db.session.begin_nested():
+            if is_update:
+                if not is_authors and not current_app.config.get('FEATURE_FLAG_ENABLE_MERGER', False):
+                    obj.log.info(
+                        'skipping update record, feature flag ``FEATURE_FLAG_ENABLE_MERGER`` is disabled.'
+                    )
+                    return
+
+                record = InspireRecord.get_record(obj.extra_data['head_uuid'])
+                obj.data['control_number'] = record['control_number']
+                record.clear()
+                record.update(obj.data, files_src_records=[obj])
+
+            else:
+                # Skip the files to avoid issues in case the record has already pid
+                # TODO: remove the skip files once labs becomes master
+                record = InspireRecord.create(obj.data, id_=None, skip_files=True)
+                # Create persistent identifier.
+                # Now that we have a recid, we can properly download the documents
+                record.download_documents_and_figures(src_records=[obj])
+
+                obj.data['control_number'] = record['control_number']
+                # store head_uuid to store the root later
+                obj.extra_data['head_uuid'] = str(record.id)
+
+            record.commit()
+            obj.save()
+    else:
+        store_record_inspirehep_api(obj, eng, is_update, is_authors)
 
-    with db.session.begin_nested():
-        if is_update:
-            if not is_authors and not current_app.config.get('FEATURE_FLAG_ENABLE_MERGER', False):
-                obj.log.info(
-                    'skipping update record, feature flag ``FEATURE_FLAG_ENABLE_MERGER`` is disabled.'
-                )
-                return
 
-            record = InspireRecord.get_record(obj.extra_data['head_uuid'])
-            obj.data['control_number'] = record['control_number']
-            record.clear()
-            record.update(obj.data, files_src_records=[obj])
+@with_debug_logging
+def store_record_inspirehep_api(obj, eng, is_update, is_authors):
+    """Saves record through inspirehep api by posting/pushing record to proper endpoint
+     in inspirehep"""
+    headers = {
+        "content-type": "application/json",
+        "Authorization": "Bearer {token}".format(
+            token=current_app.config['AUTHENTICATION_TOKEN']
```

Comment:
make the config key name less generic? `INSPIREHEP_AUTHENTICATION_TOKEN`?

Reply:
We have already the same key name for Bearer token in inspirehep. We should keep it with the same name in both projects.

Reply:
should we? in one case it's its own token, in the other case it's a foreign token. As long as it's clear what this authentication token is.